headers.h

```diff
@@ -11,15 +11,30 @@
 #define HAVE_SETSID
 
 /* OS-specific includes */
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || defined(__CYGWIN__)
+# define HAVE_UNISTD_H
+# define HAVE_SYS_IOCTL_H
 # define HAVE_SYS_SYSMACROS_H
 # define HAVE_ERROR_H
+# define HAVE_TTYNAME
```

Comment:
`unistd.h` and `ttyname` should be available on any POSIX system. So it seems that it would make more sense to test for systems without them.